File: benchmarks/suite5/gcd.c
#include <stdint.h>
#include <stdio.h>
#include "bench.h"

static int64_t gcd64(int64_t a, int64_t b) {
  while (b != 0) {
    int64_t t = a % b;
    a = b;
    b = t;
  }
  return a;
}

int main(void) {
  const int64_t n = 2500000;
  int64_t acc = 0;
  for (int64_t i = 1; i <= n; i++) {
    int64_t a = i * 9973;
    int64_t b = i * 1237 + 42;
    acc += gcd64(a, b);
  }
  suite5_print_i64(acc);
  return 0;
}

File: benchmarks/suite5/gcd.zig
const bench = @import("bench.zig");
const clib = @cImport({
    @cInclude("stdio.h");
});

fn gcd64(a: i64, b: i64) i64 {
    var x = a;
    var y = b;
    while (y != 0) {
        const t = @rem(x, y);
        x = y;
        y = t;
    }
    return x;
}

pub fn main() void {
    const n: i64 = bench.opaqueI64(2500000);
    var acc: i64 = 0;
    var i: i64 = 1;
    while (i <= n) : (i += 1) {
        const a = i * 9973;
        const b = i * 1237 + 42;
        acc += gcd64(a, b);
    }
    bench.printI64(acc);
}
